### PR TITLE
Add PR templates, workflows, and labeler config

### PR DESCRIPTION
**Pull request by ashleyshaw**
Introduces multiple pull request templates for common PR types, adds workflows for automated labeling and project field syncing, and updates labeler.yml for improved label mapping. Also adds a .github README with setup instructions, replaces label.yml with labels-issues-prs.yml, and updates .gitignore and vendor tooling.

---
name: "Default PR"
about: "General change; use for features, improvements, refactors when no specific template fits"
title: "change: <short summary>"
labels: ["status:needs-review"]
---

## Summary
**Linked issues**: Closes #<id> (and/or) Relates to #<id>

## Changes
- <bullet list of notable changes>

## Screenshots / UI (if applicable)
<add images or remove section>

## Test Notes
- [ ] Steps to verify (browsers/devices):
- [ ] Edge cases covered:

## Risk & Rollback
- Risk level: Low / Medium / High
- Rollback plan: <how to revert / flags / migrations>

---
### Checklist (Global DoD / PR)
- [ ] All AC met and demonstrated
- [ ] Tests added/updated (unit/E2E as appropriate)
- [ ] A11y considerations addressed where relevant
- [ ] Docs/readme/changelog updated (if user-facing)
- [ ] Security/perf impact reviewed where relevant
- [ ] Code/design reviews approved
- [ ] CI green; linked issues closed; release notes prepared (if shipping)

## Summary by CodeRabbit

- Documentation
  - Added comprehensive .github README describing Projects automation, setup, security and troubleshooting; added multiple PR templates for varied change types.
- CI
  - Added workflows to auto-label issues/PRs, enforce a single status label, nudge changelogs, and sync project metadata (Status/Priority/Type) from labels/branches; removed legacy label workflow.
- Chores
  - Refined label mappings and added vendor ignore rules.

### CODE REVIEW

File: .github/workflows/project-meta-sync.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            [[ "$HEAD" =~ ^docs?/ ]] && TYPE='Documentation'
+            [[ "$HEAD" =~ ^(chore/|build/) ]] && TYPE='Task'
+          fi
+
+          echo "status=$STATUS"   >> $GITHUB_OUTPUT
+          echo "priority=$PRIORITY" >> $GITHUB_OUTPUT
+          echo "type=$TYPE"       >> $GITHUB_OUTPUT
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
```
[the remark this answers is not in the document]
Comment (ashleyshaw) [reply .github/workflows/project-meta-sync.yml]: @coderabbitai .github/workflows/project-meta-sync.yml around lines 75 to 88: the workflow
currently inlines ${{ github.head_ref }} directly into a shell script inside
single quotes which can break or enable command injection if a branch name
contains quotes or special characters; instead, pass the head ref through an
environment variable (e.g., GH_HEAD) in the job/step env and read it inside the
script using the environment variable with proper double-quoting (e.g.,
"$GH_HEAD"), then use that quoted value for your pattern checks and assignments
so the shell never evaluates untrusted characters.
